Make radial linear more useful

gassmoeller opened the pull request:
The current 'radial linear' gravity profile is only useful for models of a full sphere (e.g. inner core), because it decreases linearly to 0 at the bottom of the domain (the gravity profile of a full sphere with constant density). Adding a simple input parameter makes it also useful for spherical shell models like mantle convection. Let me know if you prefer to keep the old value as default, I just thought it would be nice to have a reasonable profile for mantle convection as default. Below is a comparison of our currently available gravity profiles, with `ascii data` using the PREM profile (blue line) as the most realistic (the only one of the lines based on observations). 

As you see `radial linear` is not really good (a constant value of 10 would be closer to PREM than a radial linear profile with correct end points), but it can be a good model for the upper mantle. `radial earth-like` on the other hand is doing a terrible job, and I will open another pull request after this one that removes it (more discussion about that in the other PR)

![comparison](https://user-images.githubusercontent.com/7582930/38338346-13b3a602-381e-11e8-83e8-50b283ffe717.png)

jdannberg commented:
> radial earth-like on the other hand is doing a terrible job, and I will open another pull request after this one that removes it (more discussion about that in the other PR)

Yes, I totally agree! That has a huge potential to confuse people...

bangerth commented on source/gravity_model/radial.cc:
```diff
@@ -120,8 +120,8 @@ namespace aspect
-      return  (-magnitude_at_surface * p/p.norm() *
-               (1.0 - depth/this->get_geometry_model().maximal_depth()));
+      return  (-magnitude_at_surface  * p/p.norm() * (1.0 - depth/this->get_geometry_model().maximal_depth()))
+              + (-magnitude_at_bottom * p/p.norm() * (depth/this->get_geometry_model().maximal_depth()));
```
You can pull the factor `p/p.norm()` out of the sum. It is really just the unit outward vector whereas the rest of the formula determines the magnitude.

bangerth commented on source/gravity_model/radial.cc:
```diff
@@ -137,6 +137,14 @@ namespace aspect
+          prm.declare_entry ("Magnitude at bottom", "10.7",
+                             Patterns::Double (),
+                             "Magnitude of the radial gravity vector "
+                             "at the bottom of the domain. 'Bottom' means the"
```
Since this goes into latex, use a backtick for the opening one. Same below.

bangerth commented on source/gravity_model/radial.cc:
```diff
@@ -137,6 +137,14 @@ namespace aspect
+                             "Magnitude of the radial gravity vector "
+                             "at the bottom of the domain. 'Bottom' means the"
+                             "maximum depth in the chosen geometry, and for "
+                             "example represents the CMB in the 'spherical shell' "
```
I know it's a common abbreviation, but can you spell out CMB anyway?

bangerth commented on source/gravity_model/radial.cc:
```diff
@@ -137,6 +137,14 @@ namespace aspect
+                             "at the bottom of the domain. 'Bottom' means the"
+                             "maximum depth in the chosen geometry, and for "
+                             "example represents the CMB in the 'spherical shell' "
+                             "geometry, and the center of the 'sphere' geometry. "
```
-> in the case of the sphere geometry

gassmoeller commented:
Ok, addressed the comments. One of the tests had a slightly changed iteration count (probably floating point rounding differences), but that should be ok.
